src/serve/proxy/local-proxy.ts

```diff
@@ -55,8 +67,34 @@ export class LocalProxy implements ProxyInterface {
     });
   }
 
-  async cleanup(): Promise<void> {
+  private async connectWs(targetUrl: URL): Promise<URL> {
+    const proxy = httpProxy.createProxyServer({ target: targetUrl.toString() });
+
+    const proxyUrl = new URL("wss://localhost");
+    proxyUrl.port = this.port;
+
     return new Promise((resolve, reject) => {
+      this.server = https
+        .createServer({ key: this.key, cert: this.cert }, (req, res) => {
+          proxy.web(req, res);
+        })
+        .listen(this.port)
+        .on("listening", () => {
+          resolve(proxyUrl);
+        })
+        .on("upgrade", function (req, socket, head) {
+          req.headers["x-forwarded-proto"] = "https";
+          proxy.ws(req, socket, head);
+        })
+        .on("error", (e) => {
+          console.log(e);
```

Comment:
We can use `console.error` for logging as an exception.
```suggestion
          console.error(e);
```

Reply:
Thank you for a comment.
The error message is printed to the terminal even without using `console.error`, so I removed `console.log` itself.